### ISSUE TITLE
Mount Confusion

### ISSUE THREAD

**Issue by donquixote235** [arkayenro/arkinventory#994]
This is a very specific and oddball bug, but a bug nonetheless.  It involves the AI mount interface.

I rolled a Nightborne Rogue and set Arcanist's Manasaber as my mount.  Then I noticed that it was incorrectly classifying the mount as a land mount, so I went into config and I told it to reclassify it as a flying mount (which it is).  I then set AI to use flying mounts as land mounts, as well as to dismount in flight.

A couple of days later I decided to change mounts to match my xmog.  I unchecked the manasaber and changed to a Red Shado-Pan Riding Tiger for land and a Shadowblade's Crimson Omen as a flying mount.  (I was only lvl 50, but I chose the flying mount for when I dinged 60.)  Everything was working fine, and I logged out for the night.

When I came back the next day,  I attempted to mount and it used my flying mount.  Every time I tried to mount, the same thing would occur.  Looking at the tooltip for the AI mount interface on my Bazooka addon, it showed that I had one flying mount specified and two land mounts (which is wrong; it should have been one and one).

After troubleshooting, I was able to fix it with the following steps:

* Deselect both of my selected mounts
* In Config, reassign the Arcanist's Manasaber to a land mount
* Uncheck the Arcanist's Manasaber mount in my Bazooka interface (it was checked, even though I had unchecked it yesterday as a flying mount.)
* Reselect both of my mounts

Once this happened, everything worked fine.

I think the reason it kept picking my flying mount is because it saw the erroneously checked land version of the mount, said "oh wait that's a flying mount now, I need to pick a flying mount", and then since it wasn't checked as a flying mount it chose the one I did select.  Odd, yes.  Confusing, probably.  But it is what it is.

In summary, to reproduce:

1. Mark a mount as selected, then reclassify it (e.g. from land to flying)
2. Deselect the mount and select another one instead
3. Exit the game and re-enter

To fix:

1. Deselect your selected mount(s)
2. Reclassify the problem mount as it's default state
3. Deselect it (although it shouldn't have been selected in the first place)

**Comment by arkayenro** [comment]
the whole mount selection logic needs to be updated, the use flying mounts as land mounts doesnt really work the way it really should

**Comment by arkayenro** [comment]
should be fixed in [r697-alpha](https://www.wowace.com/projects/ark-inventory/files/2531977/download), it will clean up any mismatched selections, along with some other changes to mount selection

**Comment by arkayenro** [comment]
fixed in 3.07.41
